treemap/parseData.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * General data munging functionality
+ */
+
+import * as d3 from 'd3';
+import loadData from '@financial-times/load-data';
+
+/**
+ * Parses data file and returns structured data
+ * @param  {String} url Path to CSV/TSV/JSON file
+ * @return {Object}     Object containing series names, value extent and raw data object
+ */
+export function load(url, options) { // eslint-disable-line
+    return loadData(url).then((result) => {
+        const data = result.data ? result.data : result;
+        // make sure all the dates in the date column are a date object
+
+        // Automatically calculate the seriesnames excluding the "marker" and "annotate column"
+        let seriesNames = data.map(d => d.category);
+        seriesNames = seriesNames.filter((el, i) => seriesNames.indexOf(el) === i);
+
+        // Use the seriesNames array to calculate the minimum and max values in the dataset
+        const plotData = [{ name: 'treemap', children: buildTree() }];
+
+        function buildTree() {
+            const treeData = seriesNames.map(d => ({
+                name: d,
+                children: getChuldren(d),
```

Comment:
Why `Chuldren`?

Reply:
TBTH you only use that function once, could be rewritten simply as:

```js
children: data.filter(el => el.category === d)
```

Reply:
Ah cool, just my habit of putting eveything into functions